### PR TITLE
Fix "$lookup" when changing "attachedTo"

### PR DESCRIPTION
**Pull request by ghost**
# Contribution checklist

## Brief description
Запись экрана 2022-06-20 в 20 06 33

## Checklist

* [ ] - UI test added to added/changed functionality?
* [x] - Screenshot is added to PR if applicable ?
* [x] - Does the code work? Check whether function and logic are correct.
* [ ] - Does Changelog.md is updated with changes?
* [x] - Does the translations are up to date?
* [ ] - Does it well tested?
* [x] - Tested for Chrome.
* [x] - Tested for Safari.
* [x] - Go through the changed code looking for typos, TODOs, commented LOCs, debugging pieces of code, etc.
* [x] - Rebase your branch onto master and upstream branch
* [ ] - Is there any redundant or duplicate code?
* [x] - Are required links are linked to PR?
* [ ] - Does new code is well documented ?

## Related issues

fixes [TSK-129](https://front.hc.engineering/workbench%3Acomponent%3AWorkbenchApp/tracker%3Aapp%3ATracker/tracker%3Ateam%3ADefaultTeam/issues#tracker%3Acomponent%3AEditIssue%7C62ac17d31728e561171d6b58%7Ctracker%3Aclass%3AIssue%7Ccontent)

### CODE REVIEW

File: packages/query/src/index.ts [hunk 1]
```diff
@@ -730,15 +733,20 @@ export class LiveQuery extends TxProcessor implements Client {
...
+    _class: Ref<Class<Doc>>,
+    parent: string = ''
+  ): [string, string, string?][] {
+    const result: [string, string, string?][] = []
     const hierarchy = this.client.getHierarchy()
     if (lookup._id !== undefined) {
       for (const key in lookup._id) {
         const value = (lookup._id as any)[key]
-        const clazz = hierarchy.isMixin(value) ? hierarchy.getBaseClass(value) : value
+        const [valueClass, reverseLookupKey] = Array.isArray(value) ? value : [value]
```
Comment (BykhovDenis) [inline packages/query/src/index.ts]: const [valueClass, reverseLookupKey] = Array.isArray(value) ? value : [value, 'attachedTo']

File: packages/query/src/index.ts [hunk 2]
```diff
@@ -634,14 +634,17 @@ export class LiveQuery extends TxProcessor implements Client {
...
+      const [objWay, key, reverseLookupKey] = lookupWay
       for (const resDoc of docs) {
         const obj = getObjectValue(objWay, resDoc)
         if (obj === undefined) continue
         const value = getObjectValue('$lookup.' + key, obj)
         if (Array.isArray(value)) {
           if (this.client.getHierarchy().isDerived(doc._class, core.class.AttachedDoc)) {
-            if ((doc as AttachedDoc).attachedTo === obj._id) {
+            if (
+              (doc as AttachedDoc).attachedTo === obj._id &&
```
Comment (BykhovDenis) [inline packages/query/src/index.ts]: if (reverseLookupKey !== undefined) && (doc as any)[reverseLookupKey] === obj._id

File: packages/query/src/index.ts [hunk 3]
```diff
@@ -408,7 +408,7 @@ export class LiveQuery extends TxProcessor implements Client {
           let index = value.findIndex((p) => p._id === tx.objectId)
           if (this.client.getHierarchy().isDerived(tx.objectClass, core.class.AttachedDoc)) {
             const { attachedTo } = tx.operations as DocumentUpdate<AttachedDoc>
-            if (attachedTo !== undefined) {
+            if (attachedTo !== undefined && (reverseLookupKey === undefined || reverseLookupKey === 'attachedTo')) {
```
Comment (BykhovDenis) [inline packages/query/src/index.ts]:
```
if (reverseLookupKey !== undefined) {
const attachedTo = (tx.operations as any)[reverseLookupKey]
```

File: packages/query/src/index.ts (hunk 1 above)
Comment (ghost) [reply packages/query/src/index.ts]: Done.

File: packages/query/src/index.ts (hunk 2 above)
Comment (ghost) [reply packages/query/src/index.ts]: Done.

File: packages/query/src/index.ts (hunk 3 above)
Comment (ghost) [reply packages/query/src/index.ts]: Done.

**Review by haiodo — APPROVED** [review]

**Pull request merged**